sql/analyzer/optimization_rules.go

```diff
@@ -139,9 +140,10 @@ func nodeSources(node sql.Node) []string {
 	var result []string
 
 	for _, col := range node.Schema() {
-		if _, ok := sources[col.Source]; !ok {
-			sources[col.Source] = struct{}{}
-			result = append(result, col.Source)
+		source := fmt.Sprintf("%s.%s", col.DatabaseSource, col.Source)
```

Comment:
having these two use `sql.TableId` would be an easy correctness improvement

Reply:
Fixed.